Tests/DecoderTests.swift

```diff
@@ -33,11 +33,11 @@ class ValidDecoder: Decoder {
         self.toDictionary = toDictionary
     }
 
-    func decode(object: Any) throws -> Any {
-        let result = try Parser.parseProperties(json: object) { make in
-            make.propertyForKeyPath("subUInt", type: .uInt)
-            make.propertyForKeyPath("subInt", type: .int)
-            make.propertyForKeyPath("subString", type: .string)
+    func decode(_ object: Any) throws -> Any {
+        let result = try Parser.parseEntity(json: object) { schema in
```

Comment:
After looking further, looks like for the most part you've already done this. Looks like this one snuck through.